src/components/app/data/hooks/useCourseMetadata.js

```diff
@@ -45,10 +53,32 @@ export default function useCourseMetadata(queryOptions = {}) {
       if (!data) {
         return data;
       }
-      const availableCourseRuns = getAvailableCourseRuns({ course: data, lateEnrollmentBufferDays });
+      // First stage filters out any runs that are unavailable for universal reasons, such as enrollment windows and
+      // published states.
+      const basicAvailableCourseRuns = getAvailableCourseRuns({ course: data, lateEnrollmentBufferDays });
+      // Second stage filters out any *restricted* runs.
+      let restrictedRunFilter;
+      if (catalogUuid) {
```

Comment:
[nit/suggestion] I would generalize this approach a bit more to let the function handle whether or not the `catalogUuid` exist.

The `isRunUnrestrictedForCustomer` and `isRunUnrestrictedForCatalog` functions are similar enough that we could probably generalize it into a single`isRunUnrestricted` function.

Ex. `/app/data/utils.js`
``` js
function isRunUnrestricted({
  restrictedRunsAllowed,
  couresKey,
  courseRunMetadata,
  catalogUuid
  }) {
  const {
    allowedCatalogs,
    isRestricted,
  } = getAllowedCatalogsForRestrictedRun({
    restrictedRunsAllowed,
    courseKey,
    courseRunMetadata,
  });
  // Feed free to update this const name, using hasCatalogs for demo purposes
  const hasCatalogs = catalogUuid ? allowedCatalogs?.includes(catalogUuid) : allowedCatalogs?.length > 0
  return !isRestricted || hasCatalogs
}
```

This change allows a simpler implementation in the `useCourseMetadataHook`

Ex. `/app/data/utils.js`
``` js
      const basicAvailableCourseRuns = getAvailableCourseRuns({ course: data, lateEnrollmentBufferDays });
      // Second stage filters out any *restricted* runs.
      const restrictedRunsFilter = courseRunMetadata => isRunUnrestricted({
          restrictedRunsAllowed,
          courseKey,
          courseRunMetadata,
          catalogUuid,
      });
      const availableAndUnrestrictedCourseRuns = basicAvailableCourseRuns.filter(restrictedRunFilter);
```